### ISSUE TITLE
model-training

### ISSUE THREAD

**Issue by cliu238** [cliu238/comsa_claude#46]
## Objective
Implement the model training and evaluation functions using InSilicoVAModel with proper handling of the subpopulation parameter for both NULL and site-label conditions.

## Context
This task integrates the InSilicoVAModel to actually train models and evaluate their performance. The model should automatically detect and use the 'subpop' column when present.

## Requirements
- [ ] Implement `train_and_evaluate()` function
  - Initialize InSilicoVAModel with appropriate parameters
  - Handle Docker container execution
  - Calculate COD accuracy metrics
- [ ] Configure InSilicoVAModel parameters
  - auto_tune=False (for consistency)
  - n_sim=10000 (standard setting)
  - Handle subpop column detection automatically
- [ ] Implement accuracy calculation
  - Individual COD accuracy (per cause)
  - CSMF accuracy (population level)
  - Both metrics for comprehensive evaluation
- [ ] Add Docker failure handling
  - Retry logic for transient failures
  - Timeout handling (30 minutes max per model)
  - Clear error messages for debugging
- [ ] Implement result validation
  - Check predictions are valid
  - Ensure accuracy metrics are in [0, 1] range
  - Validate cause lists match between train/test

## Implementation Details
```python
def train_and_evaluate(
    train_data: pd.DataFrame,
    test_data: pd.DataFrame,
    use_subpop: bool = False
) -> float:
    """Train InSilicoVA model and evaluate accuracy."""
    
    # Initialize model - will auto-detect subpop column
    model = InSilicoVAModel(
        data_path=None,  # Use dataframe directly
        auto_tune=False,
        n_sim=10000,
        external_causes=False
    )
    
    # Log configuration
    has_subpop = 'subpop' in train_data.columns
    logging.info(f"Training with subpop: {has_subpop}")
    
    # Train model with retry logic
    max_retries = 3
    for attempt in range(max_retries):
        try:
            # Train model (Docker execution happens here)
            model.fit(train_data)
            break
        except Exception as e:
            if attempt == max_retries - 1:
                logging.error(f"Training failed after {max_retries} attempts: {e}")
                raise
            logging.warning(f"Attempt {attempt + 1} failed, retrying...")
            time.sleep(5)
    
    # Predict on test set
    predictions = model.predict(test_data)
    
    # Calculate accuracy metrics
    cod_accuracy = calculate_cod_accuracy(test_data['cod'], predictions['cod'])
    csmf_accuracy = calculate_csmf_accuracy(test_data['cod'], predictions['cod'])
    
    # Use COD accuracy as primary metric
    logging.info(f"COD Accuracy: {cod_accuracy:.3f}, CSMF Accuracy: {csmf_accuracy:.3f}")
    
    return cod_accuracy

def calculate_cod_accuracy(true_labels: pd.Series, pred_labels: pd.Series) -> float:
    """Calculate individual cause of death accuracy."""
    if len(true_labels) != len(pred_labels):
        raise ValueError("Label arrays must have same length")
    
    correct = (true_labels == pred_labels).sum()
    total = len(true_labels)
    return correct / total

def calculate_csmf_accuracy(true_labels: pd.Series, pred_labels: pd.Series) -> float:
    """Calculate cause-specific mortality fraction accuracy."""
    # Get cause distributions
    true_dist = true_labels.value_counts(normalize=True)
    pred_dist = pred_labels.value_counts(normalize=True)
    
    # Align distributions
    all_causes = set(true_dist.index) | set(pred_dist.index)
    true_dist = true_dist.reindex(all_causes, fill_value=0)
    pred_dist = pred_dist.reindex(all_causes, fill_value=0)
    
    # Calculate CSMF accuracy (1 - sum of absolute differences / 2)
    csmf_acc = 1 - (abs(true_dist - pred_dist).sum() / 2)
    return max(0, csmf_acc)  # Ensure non-negative
```

## Success Criteria
- InSilicoVA model trains successfully via Docker
- Subpop column is correctly detected and used when present
- Accuracy metrics are calculated correctly
- Docker failures are handled gracefully with retries
- Results are reproducible across runs
- Execution time per model < 30 minutes

## Estimated Effort
- Implementation: 2 hours
- Testing with Docker: 1.5 hours
- Total: 3.5 hours

**Comment by cliu238** [comment]
✅ Task completed. Model training with sub.pop parameter tested. Result: 0% improvement over baseline.
